component/module/module.go

```diff
@@ -40,15 +65,29 @@ func NewModuleComponent(o component.Options) (*ModuleComponent, error) {
 	return c, err
 }
 
+// NewModuleComponentV2 initializes a new ModuleComponent.
+func NewModuleComponentV2(o ModuleComponentOptions) (*ModuleComponent, error) {
```

Comment:
nit: Instead of V2 wonder if we can go with something more like `NewDeclareModuleComponent`